### DISCUSSION TITLE
saving retrieved objects using object's Name instead of CustomerKey

### DISCUSSION THREAD

**Discussion by dkhodorkovsky** [Accenture/sfmc-devtools#1379] [General]
Hi, 
is it possible to configure MCDEV tool to persist retrieved SFMC objects using object's Name instead of CustomerKey?

while some objects support setting both parameters, many others do not, such as Cloud pages, Journeys, etc.
Also, if there was no existing process demanding of a developer/consultant to set user friendly CustomerKeys , then by default, SFMC generates a GUID, and browsing these retrieved GUID named files in a repository is not easy/obvious.

thanks,

**Comment by JoernBerkefeld** [comment]
Sorry, there is no configuration option for that. However, there are tools to make your live easier in this regard:
Have you tried `mcdev fixKeys` yet?

The reason we use keys instead of names is the simple issue of uniqueness. Names are mostly not unique while keys are.

**Comment by dkhodorkovsky** [comment]
Hi, 

no, I have not tried this option - honestly, I am not sure how it works - it seems it works on deployment, not on retrieve.

is there plan to add ability to save files using object names?
I have requested developers to set Customer keys to be the same as Names, but it will not work for Cloud pages, Journeys, and some activities such as verification, wait, etc - as there is no place to specify Customer key explicitly.

**Reply by JoernBerkefeld** [reply]
for journeys the easiest approach is to CREATE them via mcdev. during create you CAN specify the key. It's just not possible to update existing ones. Annoying but works if you are doing a clean-up anyways.

no, we don't plan on changing that anytime soon. Aside from the aforementioned uniqueness issue we also try to stay close to how sf/sfdx are doing this - they also use keys
